app/(frontend)/auth/login/page.tsx

```diff
@@ -53,6 +78,9 @@ export default function LoginPage() {
                         variant="underlined"
                         className="w-3/4"
                         isRequired
+
+                        isInvalid={typeof state !== "string" && state?.password !== undefined}
+                        errorMessage={typeof state !== "string" && state?.password}
```

Comment:
**Refactor into function**
Also looks awfully similar to the logic in Line 82. Could you maybe find a solution which covers both of these cases?